CMSgov/onemac
# Oy2 20157 - Waiver Amendment Form

Story: https://qmacbis.atlassian.net/browse/OY2-20157
Endpoint: See github-actions bot comment

### Details

Create Waiver Amendment form for package view.

### Changes

- Created WaiverAmendmentForm and modified config as necessary

### Implementation Notes

- I reworked alot of the error/warning message handling around the ids to work as object arrays

### Test Plan

1. Login as state submitter
2. Navigate to package dashboard and select new submission -> Waiver Action
3. Verify label for Waiver Amendment card matches AC
4. Click Waiver Amendment option and verify you are routed to the new Waiver Amendment form.
5. Verify page title and sections per ACs
6. Enter an existing unapproved (any status other than approved) waiver number into Existing Waiver Number to Amend field
7. Verify error message per AC
8. Enter an existing approved waiver into Existing Waiver Number to Amend field
9. Verify no errors
10. Enter an invalid format waiver number into the 1915(b) Waiver Amendment Number field
11. Verify error message per AC
12. Enter a valid format waiver number into the 1915(b) Waiver Amendment Number field
13. Verify no errors
14. Fill out remaining required fields (Waiver Authority and attachments)
15. Verify submit button becomes active
16. Click Submit
17. Verify you are navigated back to package dashboard on the waiver tab and the green success banner appears at top
